src/main/java/htsjdk/variant/variantcontext/ByteArrayAllele.java

```diff
@@ -0,0 +1,305 @@
+    //
+    // accessor routines
+    //
+    // ---------------------------------------------------------------------------------------------------------
+
+    @Override
+    public boolean isPrefixOf(final Allele other) {
+        if (other.length() < this.length()) {
+            return false;
+        } else if (other instanceof ByteArrayAllele) {
+            final ByteArrayAllele baOther = (ByteArrayAllele) other;
+            return isPrefixOfBytes(baOther.bases);
+        } else {
+            return isPrefixOfBytes(other.getBases());
+        }
+    }
+
+    private boolean isPrefixOfBytes(final byte[] otherBases) {
+        for (int i = 0; i < bases.length; i++) {
+            if (bases[i] != otherBases[i]) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /** @return true if this is the NO_CALL allele */
+    @Override
+    public boolean isNoCall()           { return isNoCall; }
+    // Returns true if this is not the NO_CALL allele
+    @Override
+    public boolean isCalled()           { return ! isNoCall(); }
+
+    /** @return true if this Allele is the reference allele */
+    @Override
+    public boolean isReference()        { return isRef; }
+
+    /** @return true if this Allele is not the reference allele */
+    @Override
+    public boolean isNonReference()     { return ! isReference(); }
+
+    /** @return true if this Allele is symbolic (i.e. no well-defined base sequence), this includes breakpoints and breakends */
+    @Override
+    public boolean isSymbolic()         { return isSymbolic; }
+
+    /** @return true if this Allele is a breakpoint ( ex: G]17:198982] or ]13:123456]T ) */
+    @Override
+    public boolean isBreakpoint()         { return Allele.wouldBeBreakpoint(bases); }
+
+    /** @return true if this Allele is a single breakend (ex: .A or A.) */
+    @Override
+    public boolean isSingleBreakend()         { return Allele.wouldBeSingleBreakend(bases); }
+
+    // Returns a nice string representation of this object
+    @Override
+    public String toString() {
+        return ( isNoCall() ? NO_CALL_STRING : getDisplayString() ) + (isReference() ? "*" : "");
+    }
+
+    /**
+     * Return the DNA bases segregating in this allele.  Note this isn't reference polarized,
+     * so the Null allele is represented by a vector of length 0
+     *
+     * @return the segregating bases
+     */
+    @Override
+    public byte[] getBases() { return isSymbolic ? EMPTY_ALLELE_BASES : bases; }
+
+    /**
+     * Return the DNA bases segregating in this allele in String format.
+     * This is useful, because toString() adds a '*' to reference alleles and getBases() returns garbage when you call toString() on it.
+     *
+     * @return the segregating bases
+     */
+    @Override
+    public String getBaseString() { return isNoCall() ? NO_CALL_STRING : new String(getBases()); }
```

Comment:
This should really specify a charset.

Reply:
Is it really necessary considering that we should not encounter non-ascii characters (bytes)... Honestly I've never dealled with mixtures of charsets so perhaps you can mention an scenario in that it would be useful to use something different to the system default ... for example... if someone runs this in Japan, will he get random kanji?

Reply:
It's probably not not strictly necessary because it's unlikely that anyone will use non-ascii bases and almost every system default will keep ascii in the same position I hope.  I think it's unlikely we'd run into any problems here, we haven't in years, but there's always someone with a really crazy setting on their computer somewhere.    It's fine if you don't want to change it here.